buffer.go

```diff
@@ -26,7 +26,7 @@ func (s *buffer) Put(name string, value float64, tags Tags, timestamp int64, agg
 	s.stdBuf = append(s.stdBuf, MetricToString(name, value, tags, timestamp, aggregations, frequency))
 	s.metricCount++
 
-	if s.metricCount >= s.flushSize {
+	if s.metricCount >= s.flushSize && !s.disableAutoFlush {
```

Comment:
Not super meaningful at this point but we could invert the order of the conditions so that if it's disabled it doesn't even need to check the other condition. Same applies to the other conditions you've updated.